### ISSUE TITLE
dosbox-x sound under win98 when emulate fast cycle

### ISSUE THREAD

**Issue by l12436** [joncampbell123/dosbox-x#961]
**Describe the bug**
when I change the cpu cycle to "fixed 77000", the sound will become horrible.

**To Reproduce**
1. open config of dosbox
2. change cycles to "cycles=fixed 77000"
3. "run win98 in dosbox-x"
4. Sound will horrible when starting win98

**Expected behavior**
Sound should be fine even with faster cycles

**Environment (please complete the following information):**
 - Ubuntu 18.04
 - DOSBox-X rSHA1 "d225264b3d1df9d7194e114059458d2c23f5940a"
 - Config [https://pastebin.com/a9RSmaxu](url)

**Additional context**

**Comment by joncampbell123** [comment]
I'm sorry but sound is synchronous with emulation speed. If emulation cannot run at full speed neither can the sound.

**Comment by l12436** [comment]
> I'm sorry but sound is synchronous with emulation speed. If emulation cannot run at full speed neither can the sound.

so that mean the emulation is too fast, that my computer can not handle ?

**Comment by joncampbell123** [comment]
Emulation either runs at full speed, or slower than full speed. It *can* run faster than full speed if you have "speedlock" on.

**Comment by joncampbell123** [comment]
So it may be that you have a cycles count higher than what your Win98 system can handle.

**Comment by l12436** [comment]
> So it may be that you have a cycles count higher than what your Win98 system can handle.

OK，thanks for reply, I want to be faster because one of my game is running slow on certain screen.
that mean It maybe the fastest I can get on win98

**Comment by joncampbell123** [comment]
You might try using the menus to use the dynamic core instead of normal core.

**Comment by l12436** [comment]
> You might try using the menus to use the dynamic core instead of normal core.

yep, I am trying. although the text of the game is still a bit slower. but that maybe the fastest that the win98 can get.
thanks for help.

**Comment by l12436** [comment]
> No problem. I'm happy to hear that there are builds that run under Windows 98!
> 
> You're using the MinGW low-end builds I assume?

NO, I am using Linux. I build dosbox-x on my own. It should be the lastest version when I built.
Currently I found the other two way to execute win98, but dosbox-x is so far best. pcem seem not work very well under linux.

**Comment by l12436** [comment]
> No problem. I'm happy to hear that there are builds that run under Windows 98!
> 
> You're using the MinGW low-end builds I assume?

according to your reply, I may need to explain my environment.
I am using Linux. So far I use dosbox-x to run win95, win98, install msdos 7.10. 
Most of the time is performance issue.

**Comment by joncampbell123** [comment]
Misread that, sorry!

DOSBox-X is a bit too slow on current hardware to run Win98 in realtime, there's too much going on in the OS including kernel streaming audio.

As for Windows ME, make sure not to enable IDE emulation. There's a strange bug with DOSBox-X IDE emulation and Windows ME where Windows ME's IDE driver will conflict with INT 13h and hang at startup.

**Comment by joncampbell123** [comment]
One more thing: There's a known emulation error with MMX extensions that affects audio playback in Windows 98. Don't use Pentium MMX emulation with Windows 98.

**Comment by l12436** [comment]
> Misread that, sorry!
> 
> DOSBox-X is a bit too slow on current hardware to run Win98 in realtime, there's too much going on in the OS including kernel streaming audio.
> 
> As for Windows ME, make sure not to enable IDE emulation. There's a strange bug with DOSBox-X IDE emulation and Windows ME where Windows ME's IDE driver will conflict with INT 13h and hang at startup.

OK，so the problem is in streaming audio, so that is actually a BUG or something weird going on. 
"IDE emulation" is this a default enable option ?

> One more thing: There's a known emulation error with MMX extensions that affects audio playback in Windows 98. Don't use Pentium MMX emulation with Windows 98.

I have notice that too. a weird sound will generate with MMX

**Comment by joncampbell123** [comment]
Windows 98 has a kernel-mode audio mixing system that's always running. There's also a lot more services running at once than in Windows 95 which is why it runs slower.

IDE emulation is NOT enabled by default, you have to enable it to use it.

**Comment by l12436** [comment]
> Windows 98 has a kernel-mode audio mixing system that's always running. There's also a lot more services running at once than in Windows 95 which is why it runs slower.
> 
> IDE emulation is NOT enabled by default, you have to enable it to use it.

OK. 

So If I want to run the game faster. use WIN95 is better than WIN98 ?

**Comment by l12436** [comment]
> Windows 95 OSR2 has some of the bloat and slowdown from Windows 98 as well, by the way.

that mean I need another version of ISO. I just has OSR2 ....XD

**Comment by l12436** [comment]
after I install win95, mouse did not work .....XD

**Comment by l12436** [comment]
It much better when i increase the VMemSize. but I still need faster cycles to make some certain scene smoother.

**Comment by daerlnaxe** [comment]
> One more thing: There's a known emulation error with MMX extensions that affects audio playback in Windows 98. Don't use Pentium MMX emulation with Windows 98.

Hi, i would to know if this bug is in the 0.83.2 version ?
I'm trying to make a w98 machine...

**Comment by joncampbell123** [comment]
That hasn't been fixed yet, it's an error in MMX emulation. Sound cracking happens when emulation is slower than real speed. A capture of the audio should provide a way to hear what it would sound like at normal speed.

**Comment by daerlnaxe** [comment]
i would to know, my host is a i7-4770k (8 cores 3.5Ghz) how many cycles max i can reach without have this problem ?

**Comment by joncampbell123** [comment]
It might be better to use cycles=max. I can only guess the cycle count that prevents cracking audio on your system. It varies from system to system.

**Comment by joncampbell123** [comment]
Also know that DOSBox and DOSBox-X use pure software emulation of the CPU and everything. VirtualBox uses your CPU's virtualization extensions.

**Comment by Alevam-Inc** [comment]
I am also encountering this problem and have tried everything imaginable to no avail.
